tests/db_functions/json/test_json_remove.py

```diff
@@ -0,0 +1,249 @@
+        self.assertEqual(user_preferences.settings, {"theme": "dark"})
+
+    def test_remove_nested_key(self):
+        user_preferences = UserPreferences.objects.create(
+            settings={"font": {"size": 20, "color": "red"}}
+        )
+        UserPreferences.objects.update(settings=JSONRemove("settings", "font__color"))
+        user_preferences = UserPreferences.objects.get(pk=user_preferences.pk)
+        self.assertEqual(user_preferences.settings, {"font": {"size": 20}})
+
+    def test_remove_nested_keys_to_be_empty_object(self):
+        user_preferences = UserPreferences.objects.create(
+            settings={"font": {"color": "red"}, "notifications": True}
+        )
+        UserPreferences.objects.update(
+            settings=JSONRemove("settings", "font__color"),
+        )
+        user_preferences = UserPreferences.objects.get(pk=user_preferences.pk)
+        self.assertEqual(
+            user_preferences.settings,
+            {
+                "font": {},
+                "notifications": True,
+            },
+        )
+
+    def test_remove_multiple_keys(self):
+        user_preferences = UserPreferences.objects.create(
+            settings={"font": {"size": 20, "color": "red"}, "theme": "dark"}
+        )
+        UserPreferences.objects.update(
+            settings=JSONRemove("settings", "font__color", "theme")
+        )
+        user_preferences = UserPreferences.objects.get(pk=user_preferences.pk)
+        self.assertEqual(user_preferences.settings, {"font": {"size": 20}})
+
+    def test_remove_keys_with_recursive_call(self):
+        user_preferences = UserPreferences.objects.create(
+            settings={"font": {"size": 20, "color": "red"}, "theme": "dark"}
+        )
+        UserPreferences.objects.update(
+            settings=JSONRemove(JSONRemove("settings", "font__color"), "theme")
+        )
+        user_preferences = UserPreferences.objects.get(pk=user_preferences.pk)
+        self.assertEqual(user_preferences.settings, {"font": {"size": 20}})
+
+    def test_save_on_model_field(self):
+        user_preferences = UserPreferences.objects.create(
+            settings={"theme": "dark", "font": "Arial"}
+        )
+        user_preferences.settings = JSONRemove("settings", "theme")
+        user_preferences.save()
+
+        user_preferences = UserPreferences.objects.get(pk=user_preferences.pk)
+        self.assertEqual(user_preferences.settings, {"font": "Arial"})
+
+    def test_update_or_create_not_created(self):
+        user_preferences = UserPreferences.objects.create(
+            settings={
+                "theme": {"color": "black", "font": "Arial"},
+                "notifications": {"email": False, "sms": True},
+            }
+        )
+
+        updated_user_preferences, created = UserPreferences.objects.update_or_create(
+            defaults={"settings": JSONRemove("settings", "theme__color")},
+            id=user_preferences.id,
+        )
+        self.assertIs(created, False)
```

Comment:
```suggestion
        self.assertFalse(created)
```

Reply:
Hey @pauloxnet, just wanted to make sure you've seen the Django style [guide](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/) suggesting `assertIs` in most cases. (Speaking for myself only I think we can use common sense and let PR authors deviate from that occasionally where it makes sense, but I'd probably stop short of asking for changes going the other way.)

Reply:
If the value is controlled by us (like in this case) we should leave `assertIs(..., True)/assertIs(..., False)` instead of `assertTrue()/assertFalse()`.

Reply:
You're right, thanks for pointing this.

I'll close these suggestions and maybe I'll open a thread to propose some improvements for the style guide.